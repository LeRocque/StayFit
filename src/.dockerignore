.env
.git
.vscode
.npmrc
yarn.lock
