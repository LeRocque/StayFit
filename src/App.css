#root-app {
  height: 100vh;
}

