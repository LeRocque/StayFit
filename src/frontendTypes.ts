export type SignupModalProps = {
  handleModal: (value: void) => void;
};

export type UserWorkoutsTypes = {
  workout_id: number;
  workoutname: string;
  user_id: string;
  muscletarget: string;
  weight: string;
  reps: string;
};

export type WorkoutModalProps = {
  userId: string | undefined;
  handleWorkoutModal: (value: void) => void;
};

export type EditWorkoutModalProps = {
  workout_id: number;
  handleEditModal: (value: void) => void;
};

export type WorkoutState = {
  workouts: {
    workouts: {
      find: (
        callback: (workout: UserWorkoutsTypes) => boolean
      ) => UserWorkoutsTypes | undefined;
    };
  };
};
