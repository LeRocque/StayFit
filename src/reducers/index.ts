// import { combineReducers } from "@reduxjs/toolkit";
// import usersReducer from "./usersReducer";
// import workoutsReducer from "./workoutsReducer";

// const reducers = combineReducers({
//   users: usersReducer,
//   workouts: workoutsReducer,
// });

// export default reducers;
