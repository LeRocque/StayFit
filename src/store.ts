import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import rootReducer from "./reducers";
import { persistStore, persistReducer, PERSIST } from "redux-persist";
import storage from "redux-persist/lib/storage";

const persistConfig = {
  key: "root",
  storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [PERSIST],
      },
    }),
});

export const persistor = persistStore(store);

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
