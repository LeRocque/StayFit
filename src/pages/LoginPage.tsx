import { ChangeEvent, FormEvent, useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";
import {
  loginUserActionCreator,
  setImagesActionCreator,
} from "../actions/actions";
import { SignupModal } from "../components/SignupModal";
import { useAppDispatch } from "../hooks";
import { ReturnedUserId, WorkoutImages } from "../frontendTypes";

const LoginPage = () => {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [showSignupModal, setShowSignupModal] = useState(false);

  const navigate = useNavigate();
  const dispatch = useAppDispatch();

  useEffect(() => {
    const getWorkoutImages = async () => {
      try {
        const result = await fetch("/workout/images");
        const images = (await result.json()) as WorkoutImages;
        dispatch(setImagesActionCreator(images));
      } catch (err) {
        console.error(err);
      }
    };
    getWorkoutImages();
  }, [dispatch]);

  // console.log("workoutImages are:", workoutImages);

  const handleModal = () => {
    setShowSignupModal(!showSignupModal);
  };

  const handleSetUsername = (e: ChangeEvent<HTMLInputElement>) => {
    setUsername(e.target.value);
  };
  const handleSetPassword = (e: ChangeEvent<HTMLInputElement>) => {
    setPassword(e.target.value);
  };

  const handleSubmit = async (e: FormEvent) => {
    e.preventDefault();
    try {
      const response = await fetch("/user/login", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        credentials: "include",
        body: JSON.stringify({
          username: username,
          password: password,
        }),
      });
      setUsername("");
      setPassword("");
      if (response.ok) {
        const returnedUserId = (await response.json()) as ReturnedUserId;
        const returnedId = returnedUserId.user_id;
        dispatch(loginUserActionCreator(username));
        navigate(`/home/${returnedId}`);
      } else if (response.status === 401) {
        alert("Invalid username or password");
      } else if (response.status === 400) {
        alert("both fields required");
      }
    } catch (err) {
      console.error(err);
    }
  };
  return (
    <div className="h-100 my-0 mx-auto flex w-1/4">
      <div id="login-container" className="m-10 w-full p-7 text-center">
        <div className="text-slate-600">Welcome to Workout Tracker!</div>
        <form
          className="mb-10 mt-3 flex flex-col items-center justify-center rounded-lg bg-slate-600 px-3 py-10"
          onSubmit={handleSubmit}
        >
          <input
            className="search-input"
            type="text"
            required
            name="username"
            placeholder="Enter your username"
            value={username}
            onChange={handleSetUsername}
          />
          <input
            className="search-input"
            type="password"
            required
            name="password"
            placeholder="Enter your password"
            value={password}
            onChange={handleSetPassword}
          />
          <button className="button-theme" type="submit">
            Login
          </button>
        </form>
        <button className="button-theme" onClick={handleModal}>
          Signup
        </button>
      </div>
      {showSignupModal && <SignupModal handleModal={handleModal} />}
    </div>
  );
};

export default LoginPage;
