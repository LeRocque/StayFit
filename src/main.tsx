import React, { createRoot } from "react-dom/client";
import App from "./App";
import { Provider } from "react-redux";
import { PersistGate } from "redux-persist/integration/react";
import { persistor, store } from "./store";
import { BrowserRouter } from "react-router-dom";

createRoot(document.getElementById("root") as HTMLElement).render(
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </PersistGate>
  </Provider>,
);
