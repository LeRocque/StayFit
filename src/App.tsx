import { Routes, Route, Navigate } from "react-router-dom";
import { LoginPage } from "./pages/LoginPage";
import { HomePage } from "./pages/HomePage";
import { RootState } from "./store";
import { useAppSelector } from "./hooks";
import "./style.css";

const App = () => {
  return (
    <div
      id="root-app"
      className="background-red m-0 bg-gradient-to-b from-blue-200 via-blue-400 to-blue-900 p-0"
    >
      <Routes>
        <Route path="/" element={<LoginPage />} />
        <Route path="/home/:userId" element={<ProtectedRoute />} />
      </Routes>
    </div>
  );
};

const ProtectedRoute = () => {
  const isAuthenticated = useAppSelector(
    (state: RootState) => state.users.isAuthenticated,
  );
  if (isAuthenticated) {
    return <HomePage />;
  } else {
    alert("You must be logged in to view that page");
    return <Navigate to={"/"} />;
  }
};

export default App;
