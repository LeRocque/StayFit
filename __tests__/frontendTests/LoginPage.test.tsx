import { expect, test } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../../src/store";
import LoginPage from "../../src/pages/LoginPage";
import { BrowserRouter } from "react-router-dom";

test("LoginPage should match it's snapshot", () => {
  const loginPage = render(
    <Provider store={store}>
      <BrowserRouter>
        <LoginPage />
      </BrowserRouter>
    </Provider>,
  );

  expect(loginPage).toMatchSnapshot();
  loginPage.unmount();
});

test("LoginPage should show signup modal form when signup button clicked", () => {
  const loginPage = render(
    <Provider store={store}>
      <BrowserRouter>
        <LoginPage />
      </BrowserRouter>
    </Provider>,
  );
  const notClicked = loginPage.queryByTestId("signup-form");
  expect(notClicked).toBeNull();
  const signupButton = loginPage.getByTestId("signup-button");
  fireEvent.click(signupButton);
  const signupForm = loginPage.getByTestId("signup-form");
  expect(signupForm).toBeDefined();
});
