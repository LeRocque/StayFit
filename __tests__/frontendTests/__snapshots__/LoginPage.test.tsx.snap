// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginPage should match it's snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex h-screen flex-col items-center justify-center  bg-gradient-to-br from-blue-200 via-blue-500 to-blue-700"
      >
        <div
          class="mb-10 rounded-md bg-white p-4 text-center shadow-lg"
        >
          <h1
            class="text-3xl font-semibold text-blue-800"
          >
            Welcome to Workout Tracker!
          </h1>
          <form
            class="mb-5 mt-3 flex flex-col items-center justify-center rounded-lg bg-slate-400 px-3 py-10"
          >
            <input
              class="search-input"
              name="username"
              placeholder="Enter your username"
              required=""
              type="text"
              value=""
            />
            <input
              class="search-input"
              name="password"
              placeholder="Enter your password"
              required=""
              type="password"
              value=""
            />
            <button
              class="button-theme"
              type="submit"
            >
              Login
            </button>
          </form>
          <button
            class="button-theme"
            data-testid="signup-button"
          >
            Signup
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex h-screen flex-col items-center justify-center  bg-gradient-to-br from-blue-200 via-blue-500 to-blue-700"
    >
      <div
        class="mb-10 rounded-md bg-white p-4 text-center shadow-lg"
      >
        <h1
          class="text-3xl font-semibold text-blue-800"
        >
          Welcome to Workout Tracker!
        </h1>
        <form
          class="mb-5 mt-3 flex flex-col items-center justify-center rounded-lg bg-slate-400 px-3 py-10"
        >
          <input
            class="search-input"
            name="username"
            placeholder="Enter your username"
            required=""
            type="text"
            value=""
          />
          <input
            class="search-input"
            name="password"
            placeholder="Enter your password"
            required=""
            type="password"
            value=""
          />
          <button
            class="button-theme"
            type="submit"
          >
            Login
          </button>
        </form>
        <button
          class="button-theme"
          data-testid="signup-button"
        >
          Signup
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
